[2025-10-18T20:50:02.623+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-18T20:50:02.667+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-18T20:45:00+00:00 [queued]>
[2025-10-18T20:50:02.683+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-18T20:45:00+00:00 [queued]>
[2025-10-18T20:50:02.685+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-10-18T20:50:02.716+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_weather_data> on 2025-10-18 20:45:00+00:00
[2025-10-18T20:50:02.729+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=284) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-10-18T20:50:02.732+0000] {standard_task_runner.py:63} INFO - Started process 287 to run task
[2025-10-18T20:50:02.727+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'flood_prediction_pipeline', 'fetch_weather_data', 'scheduled__2025-10-18T20:45:00+00:00', '--job-id', '146', '--raw', '--subdir', 'DAGS_FOLDER/flood_prediction_dag.py', '--cfg-path', '/tmp/tmpe6wq2lzn']
[2025-10-18T20:50:02.734+0000] {standard_task_runner.py:91} INFO - Job 146: Subtask fetch_weather_data
[2025-10-18T20:50:02.814+0000] {task_command.py:426} INFO - Running <TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-18T20:45:00+00:00 [running]> on host fcca755cbcb6
[2025-10-18T20:50:02.970+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='flood_prediction_pipeline' AIRFLOW_CTX_TASK_ID='fetch_weather_data' AIRFLOW_CTX_EXECUTION_DATE='2025-10-18T20:45:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-18T20:45:00+00:00'
[2025-10-18T20:50:02.974+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-18T20:50:03.021+0000] {logging_mixin.py:188} INFO - âœ… Weather data fetched: {'temp': 21.624510756351288, 'humidity': 87.56392164723744, 'pressure': 1027.793327738706, 'wind_speed': 4.122077473968422, 'rain_1h': 22.821245939012172, 'rain_3h': 36.7653670940072, 'rain_6h': 131.50814161287116, 'rain_24h': 84.70514890651248, 'rain_intensity': 5.107304250362139, 'temp_delta': -1.0396389761437028, 'humidity_delta': 4.376099465285648, 'year': 2025, 'month': 10, 'day': 18, 'hour': 20, 'minute': 50, 'second': 2}
[2025-10-18T20:50:03.022+0000] {python.py:237} INFO - Done. Returned value was: {'temp': 21.624510756351288, 'humidity': 87.56392164723744, 'pressure': 1027.793327738706, 'wind_speed': 4.122077473968422, 'rain_1h': 22.821245939012172, 'rain_3h': 36.7653670940072, 'rain_6h': 131.50814161287116, 'rain_24h': 84.70514890651248, 'rain_intensity': 5.107304250362139, 'temp_delta': -1.0396389761437028, 'humidity_delta': 4.376099465285648, 'year': 2025, 'month': 10, 'day': 18, 'hour': 20, 'minute': 50, 'second': 2}
[2025-10-18T20:50:03.023+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-18T20:50:03.054+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=flood_prediction_pipeline, task_id=fetch_weather_data, run_id=scheduled__2025-10-18T20:45:00+00:00, execution_date=20251018T204500, start_date=20251018T205002, end_date=20251018T205003
[2025-10-18T20:50:03.109+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-10-18T20:50:03.140+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-10-18T20:50:03.145+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
